contrib/win32/win32compat/fileio.c

```diff
@@ -421,14 +422,36 @@ createFile_flags_setup(int flags, mode_t mode, struct createFile_flags* cf_flags
 	return ret;
 }
 
+/* returns 1 if true, 0 otherwise */
+int
+file_in_chroot_jail(HANDLE handle, const char* path_utf8) {
+	/* ensure final path is within chroot */
+	wchar_t path_buf[MAX_PATH], *final_path;
+	if (GetFinalPathNameByHandleW(handle, path_buf, MAX_PATH, 0) == 0) {
+		debug3("failed to get final path of file:%s error:%d", path_utf8, GetLastError());
+		return 0;
+	}
+	final_path = path_buf + 4;
+	to_wlower_case(final_path);
+	if ((wcslen(final_path) < wcslen(chroot_pathw)) ||
+		memcmp(final_path, chroot_pathw, 2 * wcslen(chroot_pathw)) != 0 ||
+		final_path[wcslen(chroot_pathw)] != '\\') {
+		debug4("access denied due to attempt to escape chroot jail");
```

Comment:
Having it as debug() will help ssh admins to run some metrics and figureout the potential attacks..